pl_examples/full_examples/semantic_segmentation/semseg.py

```diff
@@ -0,0 +1,151 @@
+        if self.split == 'train':
+            self.img_path = os.path.join(self.root, 'training/image_2')    
+            self.mask_path = os.path.join(self.root, 'training/semantic')
+        else : 
+            self.img_path = os.path.join(self.root, 'testing/image_2')
+            self.mask_path = None
+
+        self.transform = transform
+        
+        self.img_list = self.get_filenames(self.img_path)
+        if self.split == 'train':
+            self.mask_list = self.get_filenames(self.mask_path)
+        else :
+            self.mask_list = None
+        
+    def __len__(self):
+        return(len(self.img_list))
+    
+    def __getitem__(self, idx):
+        img = Image.open(self.img_list[idx])
+        img = img.resize(self.img_size)
+        img = np.array(img)
+        
+        if self.split == 'train' : 
+            mask = Image.open(self.mask_list[idx]).convert('L')
+            mask = mask.resize(self.img_size)
+            mask = np.array(mask)
+            mask = self.encode_segmap(mask)
+        
+        if self.transform :
+            img = self.transform(img)
+            
+        if self.split == 'train' :
+            return img, mask
+        else :
+            return img
+    
+    def encode_segmap(self, mask):
+        '''
+        Sets void classes to zero so they won't be considered for training
+        '''
+        for voidc in self.void_labels :
+            mask[mask == voidc] = self.ignore_index
+        for validc in self.valid_labels :
+            mask[mask == validc] = self.class_map[validc]
+        return mask
+    
+    def get_filenames(self, path):
+        files_list = list()
+        for filename in os.listdir(path):
+            files_list.append(os.path.join(path, filename))
+        return files_list
+    
+class SegModel(pl.LightningModule):
+    def __init__(self, hparams):
+        super(SegModel, self).__init__()
+        self.hparams = hparams
+        self.root_path = hparams.root
+        self.batch_size = hparams.batch_size
+        self.learning_rate = hparams.lr
+        self.net = torchvision.models.segmentation.fcn_resnet50(pretrained = False, progress = True, num_classes = 19)
+        self.transform = transforms.Compose([
+            transforms.ToTensor(),
+            transforms.Normalize(mean = [0.35675976, 0.37380189, 0.3764753], std = [0.32064945, 0.32098866, 0.32325324])
+        ])
+        self.trainset = KITTI(self.root_path, split = 'train', transform = self.transform)
+        self.testset = KITTI(self.root_path, split = 'test', transform = self.transform)
+        
+    def forward(self, x):
+        return self.net(x)
+    
+    def training_step(self, batch, batch_nb) :
+        img, mask = batch
+        img = img.float()
+        mask = mask.long()
+        out = self.forward(img)
+        loss_val = F.cross_entropy(out['out'], mask, ignore_index = 250)
+        return {'loss' : loss_val}
+    
+#     def test_step(self, batch, batch_nb):
```

Comment:
why is this commented? if it not needed, pls remove it...

Reply:
This will be needed for the inference usage you mentioned in another comment. But, I haven't been able to get it to work yet.